src/mpid/ch4/shm/posix/posix_impl.h

```diff
@@ -18,8 +18,12 @@
 #include "posix_eager.h"
 #include "posix_eager_impl.h"
 
-#undef FUNCNAME
-#define FUNCNAME nothing
+/* FIXME: This should not work ...
+ * MPIR_FUNC_VERBOSE_STATE_DECL(a) etc uses #a to turn a into a string
+ * This macro is uses e.g. BEGIN_FUNC(__func__), but __func__ is a string
+ * So #a (as I quickly tested) simply become "__func__".
+ * Reviewers: please verify ...
+ */
 #define BEGIN_FUNC(FUNCNAME)                    \
     MPIR_FUNC_VERBOSE_STATE_DECL(FUNCNAME);     \
     MPIR_FUNC_VERBOSE_ENTER(FUNCNAME);
```

Comment:
Same thing here. Dump this macro.

Reply:
I agree on dumping these two macros. Shall we just revert the usage of `BEGIN_FUNC` and `END_FUNC` back to `MPIR_FUNC_VERBOSE_{STATE_DECL,ENTER,EXIT}`?

Reply:
Yep.